addons/binding/org.openhab.binding.autelis/src/main/java/org/openhab/binding/autelis/discovery/AutelisDiscoveryService.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2014 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.autelis.discovery;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.commons.io.IOUtils;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.autelis.AutelisBindingConstants;
+
+/**
+ * 
+ * Discovery Service for Autelis Pool Controllers.
+ * @author Dan Cunningham
+ *
+ */
+public class AutelisDiscoveryService extends AbstractDiscoveryService{
+
+	private final Logger logger = LoggerFactory.getLogger(AutelisDiscoveryService.class);
+	 
+	private static String DEFAULT_NAME = "poolcontrol";
+	
+	public AutelisDiscoveryService()
+			throws IllegalArgumentException {
+		super(AutelisBindingConstants.SUPPORTED_THING_TYPES_UIDS, 10);
+	}
+	
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return AutelisBindingConstants.SUPPORTED_THING_TYPES_UIDS;
+    }
+    
+    @Override
+    protected void startBackgroundDiscovery() {
+        new Thread(new Runnable() {
```

Comment:
Instead of creating a new thread, could you please use the available scheduler field from AbstractDiscoveryService?